modules/exampleauth/src/Auth/Source/UserPass.php

```diff
@@ -48,6 +48,8 @@ public function __construct(array $info, array $config)
                 throw new Exception(
                     'Invalid <username>:<password> for authentication source ' . $this->authId . ': ' . $userpass
                 );
+            } elseif ($userpass === 'authproc') {
```

Comment:
This is hackish and I don't agree with it

Reply:
The "hackishness" of this particular part of this PR is fixed in https://github.com/simplesamlphp/simplesamlphp/pull/2003

Reply:
Is this still needed now that 2003 is merged? If people are wanting to define authproc here they should probably have moved to having the user:pass inside the `users` subitem?

Reply:
reading below, me comment above seems less interesting than a the big picture comments further down in the PR.

Reply:
Yeah. This particular part of the PR can be removed. I'd rework the PR based on feedback if it was a chance of being merged.

Right now, I'm working around having this one merged by having all my authprocs that only apply to one IdP globally instead of with that IdP that they should apply to. Less secure, messier config, etc.

I'm not sure why per-SP authprocs are allowed, but per-IdP authprocs aren't allowed (or desirable).